### PR TITLE
itest: prepare itest for major refactor

### PR DESCRIPTION
**Pull request by yyforyongyu**
This PR has commits pulled out from #5939 for easier reviewing, only the last 6 commits are new.

Depends on #5756.

### CODE REVIEW

File: lntest/harness_node.go [hunk 1]
```diff
@@ -2007,3 +1881,94 @@ func (hn *HarnessNode) getChannelPolicies(include bool) policyUpdateMap {
...
+// renameFile is a helper to rename (log) files created during integration
+// tests.
+func renameFile(fromFileName, toFileName string) {
+	err := os.Rename(fromFileName, toFileName)
+	if err != nil {
+		fmt.Printf("could not rename %s to %s: %v\n",
+			fromFileName, toFileName, err)
+	}
+}
+func getFinalizedLogFilePrefix(hn *HarnessNode) string {
```
Comment (guggero) [inline lntest/harness_node.go:1835]: nit: add empty line and godoc.

File: lntest/harness_node.go [hunk 2]
```diff
@@ -768,6 +684,28 @@ func (hn *HarnessNode) start(lndBinary string, lndError chan<- error,
 		return err
 	}
 
+	return nil
```
Comment (guggero) [inline lntest/harness_node.go:626]: If we don't keep track of the PID file, do we even need to write it in the first place?

File: lntest/harness_node.go [hunk 3]
```diff
@@ -317,41 +317,50 @@ type HarnessNode struct {
 	// node and the outpoint.
 	policyUpdates policyUpdateMap
 
+	// conn is the underlying connection to the grpc endpoint of the node.
+	conn *grpc.ClientConn
```
Comment (guggero) [inline lntest/harness_node.go:319]: Move this to `RPCClients` struct as well?

File: lntest/harness_node.go [hunk 4]
```diff
@@ -989,17 +1017,36 @@ func (hn *HarnessNode) waitTillServerStarted() error {
...
+		Router:           routerrpc.NewRouterClient(c),
+		WalletKit:        walletrpc.NewWalletKitClient(c),
+		WalletUnlocker:   lnrpc.NewWalletUnlockerClient(c),
+		Watchtower:       watchtowerrpc.NewWatchtowerClient(c),
+		WatchtowerClient: wtclientrpc.NewWatchtowerClientClient(c),
+		Signer:           signrpc.NewSignerClient(c),
+		State:            lnrpc.NewStateClient(c),
+	}
 
+	hn.rpc = rpcClients
```
Comment (guggero) [inline lntest/harness_node.go:927]: nit: assign new struct directly?

**Review by guggero — APPROVED** [review]
Nice refactor, LGTM :tada:

File: lntest/harness_node.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/harness_node.go:1835]: Done!

File: lntest/harness_node.go (hunk 3 above)
Comment (yyforyongyu) [reply lntest/harness_node.go:319]: Done!

File: lntest/harness_node.go (hunk 4 above)
Comment (yyforyongyu) [reply lntest/harness_node.go:927]: Done!

File: lntest/harness_node.go [hunk 5]
```diff
@@ -190,28 +186,32 @@ func (cfg NodeConfig) genArgs() []string {
...
+		fmt.Sprintf("--tlskeypath=%v", cfg.TLSKeyPath),
+		fmt.Sprintf("--configfile=%v", cfg.DataDir),
+		fmt.Sprintf("--adminmacaroonpath=%v", cfg.AdminMacPath),
+		fmt.Sprintf("--readonlymacaroonpath=%v", cfg.ReadMacPath),
+		fmt.Sprintf("--invoicemacaroonpath=%v", cfg.InvoiceMacPath),
+		fmt.Sprintf("--trickledelay=%v", trickleDelay),
+		fmt.Sprintf("--profile=%d", cfg.ProfilePort),
+		fmt.Sprintf("--caches.rpc-graph-cache-duration=0"),
+	}
+	args = append(args, nodeArgs...)
```
Comment (positiveblue) [inline lntest/harness_node.go:214]: better :+1:

**Review by positiveblue — APPROVED** [review]
Thank you @yyforyongyu, test suites improvements keep adding up :champagne:

**Comment by guggero** [comment]
The RPC middleware itest seems to fail consistently now. Should I take a look or do you already have a hunch what's causing the failure?

**Comment by yyforyongyu** [comment]
> The RPC middleware itest seems to fail consistently now. Should I take a look or do you already have a hunch what's causing the failure?

If you are talking about this one,
```
lnd_rpc_middleware_interceptor_test.go:407: 
        	Error Trace:	lnd_rpc_middleware_interceptor_test.go:407
        	            				lnd_rpc_middleware_interceptor_test.go:125
        	Error:      	Received unexpected error:
        	            	failed to subscribe to state: rpc error: code = Canceled desc = grpc: the client connection is closing
```

Yeah it's been fixed in the last PR.

**Comment by guggero** [comment]
The RPC interceptor test still seems to fail with `failed to subscribe to state: rpc error: code = Canceled desc = grpc: the client connection is closing`.
I'm not sure what you mean with "Yeah it's been fixed in the last PR.". Do you mean last commit? Or another PR and this needs to be rebased?

**Comment by yyforyongyu** [comment]
> The RPC interceptor test still seems to fail with `failed to subscribe to state: rpc error: code = Canceled desc = grpc: the client connection is closing`. I'm not sure what you mean with "Yeah it's been fixed in the last PR.". Do you mean last commit? Or another PR and this needs to be rebased?

Sorry, I meant in the final PR #5940, in particular [this commit](https://github.com/lightningnetwork/lnd/pull/5940/commits/056b48b1d607316d51ba40b7dcd8320f28628963).

This PR contains some commits taken from #5939. I'd say it's an "intermediate" PR that exists only for easier reviewing.

**Comment by guggero** [comment]
Ah I see. So should we wait with merging this PR? Or add that commit to it so we don't fully break the itests?

**Comment by yyforyongyu** [comment]
> Ah I see. So should we wait with merging this PR? Or add that commit to it so we don't fully break the itests?

Yeah let's wait. Actually I'm kinda stuck here😂 and opinions are appreciated. These itest-related fix has gone quite far than I originally anticipated, and it's challenging to, say, cherry-pick some commits from later PRs so we have a clean build in each PR because the changes are chained. I'm thinking we could focus the itest build results in the final PR just to make sure the flakes are fixed. Meanwhile, we could somehow loosen checking the itest build results in the intermediate PRs? And for the following PR #5939, do you think it's too large? Maybe I should break it down even more?

**Comment by guggero** [comment]
I see, it's getting tricky indeed. But I'd rather not keep the itests in a state where they _all_ fail reliably, that would render us completely blind to new bugs being introduced. Instead, I was able to extract the fix into a single commit that you can add to this PR so we can merge it (if nothing else pops up): https://github.com/guggero/lnd/commit/dba9ade7af3ff382df4250d4b084e58e5010d933

**Comment by yyforyongyu** [comment]
> I was able to extract the fix into a single commit that you can add to this PR so we can merge it

Cool cherry-picked.

**Pull request merged**